primitives/block-builder/src/lib.rs

```diff
@@ -20,18 +20,24 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use sp_inherents::{CheckInherentsResult, InherentData};
-use sp_runtime::{traits::Block as BlockT, ApplyExtrinsicResult};
+use sp_runtime::{
+	legacy::before_v6::ApplyExtrinsicResult as ApplyExtrinsicResultBeforeV6,
+	traits::Block as BlockT, ApplyExtrinsicResult,
+};
 
 sp_api::decl_runtime_apis! {
 	/// The `BlockBuilder` api trait that provides the required functionality for building a block.
-	#[api_version(5)]
+	#[api_version(6)]
 	pub trait BlockBuilder {
 		/// Apply the given extrinsic.
 		///
 		/// Returns an inclusion outcome which specifies if this extrinsic is included in
 		/// this block or not.
 		fn apply_extrinsic(extrinsic: <Block as BlockT>::Extrinsic) -> ApplyExtrinsicResult;
 
+		#[changed_in(6)]
+		fn apply_extrinsic(extrinsic: <Block as BlockT>::Extrinsic) -> ApplyExtrinsicResultBeforeV6;
```

Comment:
While this is right, you will also need to change the code that calls this function to handle this.

Reply:
Do you mean I have to implement `apply_extrinsic_before_version_6`? I have no idea what I would need to change otherwise...

Reply:
No, you need to use `apply_extrinsic_before_version_6`.

So currently everywhere where we have `apply_extrinsic` it needs to be changed to:
```
let runtime_api = client.runtime_api();

if runtime_api.api_version::<RuntimeApi>(at) < 6 {
    runtime_api.apply_extrinsic_before_version_6(at, tx).map(convert_error)
} else {
    runtime_api.apply_extrinsic(at, tx)
}
```